src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/UidValueResolver.php

```diff
@@ -35,6 +35,7 @@ public function supports(Request $request, ArgumentMetadata $argument): bool
 
     public function resolve(Request $request, ArgumentMetadata $argument): array
     {
+        /** @var class-string<AbstractUid> $uidClass */
```

Comment:
If we change the if statement like this, static analysis should be able to derive this without inline PHPdoc:

```diff
         if ($argument->isVariadic()
              || !\is_string($value = $request->attributes->get($argument->getName()))
              || null === ($uidClass = $argument->getType())
-             || !is_subclass_of($argument->getType(), AbstractUid::class, true)
+             || !is_subclass_of($uidClass, AbstractUid::class, true)
          ) {
```

Reply:
Yes, you're right, done